clinica/pipelines/statistics_surface/_model.py

```diff
@@ -0,0 +1,728 @@
+        stored in a `CorrectedPValueResults` instance.
+    """
+
+    def __init__(
+        self,
+        coefficients: np.ndarray,
+        tstats: np.ndarray,
+        uncorrected_p_values: PValueResults,
+        fdr: np.ndarray,
+        corrected_p_values: CorrectedPValueResults,
+    ) -> None:
+        self.coefficients = coefficients
+        self.TStatistics = tstats
+        self.uncorrectedPValue = uncorrected_p_values
+        self.FDR = fdr
+        self.correctedPValue = corrected_p_values
+
+    @classmethod
+    def from_slm_model(
+        cls,
+        model: SLM,
+        mask: np.ndarray,
+        threshold_uncorrected_p_value: float,
+        threshold_corrected_p_value: float,
+    ):
+        idx = np.argwhere(np.isnan(model.t))
+        corrected_pvals = model.P["pval"]["P"]
+        corrected_pvals[idx] = 1.0
+        tstats = np.nan_to_num(model.t)
+        uncorrected_p_values = PValueResults.from_t_statistics(
+            tstats,
+            model.df,
+            mask,
+            threshold_uncorrected_p_value,
+        )
+        corrected_p_values = CorrectedPValueResults(
+            corrected_pvals,
+            model.P["pval"]["C"],
+            mask,
+            threshold_corrected_p_value,
+        )
+        return cls(
+            np.nan_to_num(model.coef),
+            tstats,
+            uncorrected_p_values,
+            model.Q,
+            corrected_p_values,
+        )
+
+    def to_dict(self, jsonable: bool = True):
+        """Returns the results in dict format.
+
+        Parameters
+        ----------
+        jsonable: If `True`, the output can be directly written in JSON.
+            Otherwise, it might contain non-serializable objects.
+        """
+        import inspect
+
+        json_dict = dict()
+        for attribute in inspect.getmembers(self):
+            name, value = attribute
+            if not name.startswith("_"):
+                if not inspect.ismethod(value):
+                    if hasattr(value, "to_dict"):
+                        json_dict[name] = value.to_dict(jsonable=jsonable)
+                    elif isinstance(value, np.ndarray) and jsonable:
+                        json_dict[name] = value.tolist()
+                    else:
+                        json_dict[name] = value
+        return json_dict
+
+
+class StatisticsResultsPlotter:
+    def __init__(self, output_file: PathLike, mesh: Mesh):
+        self.output_file = output_file
+        self.mesh = mesh
+        self.plotting_extension = ".png"
+        self.no_plot = {"coefficients"}  # Elements which should not be plotted
+
+    def plot(self, result: StatisticsResults, method: str):
+        plotter = self._get_plotter(method)
+        return plotter(result)
+
+    def _get_plotter(self, method: str):
+        if method == "nilearn_plot_surf_stat_map":
+            return self._plot_stat_maps
+        else:
+            raise NotImplementedError(f"Plotting method {method} is not implemented.")
```

Comment:
Do we really need dynamic dispatch? It messes with static code analysis of the `plot` message and it raises the cognitive overhead for the code.

Reply:
I solved the static analysis part. Still not sure what dynamic dispatch provides here.

Reply:
I implemented it this way as I thought we could add other plotting methods with little cost. I was thinking at other, more advanced, surface plotting functions provided by Nilearn, like [view_img_on_surf](https://nilearn.github.io/dev/modules/generated/nilearn.plotting.view_img_on_surf.html#nilearn.plotting.view_img_on_surf) for instance. We could also use plotting capabilities of Brainstat although I'm not familiar with them.
Looking at it now, I agree this might not be the best implementation. So, if you think we won't need additional plotting capabilities and this solution is too complex, I can simplify.